src/clib/lib/enkf/state_map.cpp

```diff
@@ -119,7 +119,8 @@ void StateMap::set(int index, realisation_state_enum new_state) {
     if (index < 0)
         throw exc::out_of_range("index out of range: {} < 0", index);
     if (index >= m_state.size())
-        m_state.resize(index + 1, STATE_UNDEFINED);
+        throw exc::out_of_range("index out of range: {} >= {}", index,
```

Comment:
I think we should remove these `out_of_range` ifs and let `m_state.at(index)` do the job of throwing exceptions.